controller/src/main/java/io/pravega/controller/server/security/auth/AuthorizationResource.java

```diff
@@ -101,4 +113,13 @@
      * @throws IllegalArgumentException if {@code scopeName} or {@code kvtName} are empty
      */
     String ofKeyValueTableInScope(String scopeName, String kvtName);
+
+    /**
+     * Creates a resource representation for use in authorization of actions related to the specified internal stream.
+     *
+     * @param scopeName the name of the scope
+     * @param streamName the name of the internal stream
+     * @return a string representing the resource
+     */
+    String ofInternalStream(String scopeName, String streamName);
```

Comment:
Firstly, about the differentiation between the internal vs. external streams. The intent is: 
1. To treat resource groups and watermarks as resources in their own right and distinguish them from regular streams. This is so that users can  be assigned ACLs that authorize them to perform Controller operations on them.
2. For all internal streams, enable write operations with read permissions, provided the config allows for it. 
[This ](https://github.com/pravega/pravega/wiki/PDP-46:-Read-Only-Permissions-For-Reading-Data)PDP explains the overall context in more detail. 

Secondly, for the question below
> ... would rg stream be treated as "stream" and "internal stream" interchangeably depending on the context?

No, the intent is to always authorize a resource the same way across all operations. If the code doesn't do that, it'd be a flaw. 

Does that answer your question @shiveshr?